tests/test_pipx_metadata_file.py

```diff
@@ -0,0 +1,140 @@
+    venv_dir1 = tmp_path / "venv1"
+    venv_dir1.mkdir()
+    venv_dir2 = tmp_path / "venv2"
+    venv_dir2.mkdir()
+
+    test_package1 = TEST_PACKAGE1._replace(include_apps=False)
+    test_package2 = TEST_PACKAGE1._replace(package=None)
+    test_package3 = TEST_PACKAGE1._replace(package_or_url=None)
+
+    for test_package in [test_package1, test_package2, test_package3]:
+        pipx_metadata = PipxMetadata(venv_dir1)
+        pipx_metadata.main_package = test_package
+        pipx_metadata.python_version = "3.4.5"
+        pipx_metadata.venv_args = ["--system-site-packages"]
+        pipx_metadata.injected_packages = {}
+
+        with pytest.raises(PipxError):
+            pipx_metadata.write()
+
+
+def test_package_install(monkeypatch, tmp_path, pipx_temp_env):
+    pipx_venvs_dir = pipx.constants.PIPX_HOME / "venvs"
+
+    run_pipx_cli(["install", "pycowsay"])
+    assert (pipx_venvs_dir / "pycowsay" / "pipx_metadata.json").is_file()
+
+    pipx_metadata = PipxMetadata(pipx_venvs_dir / "pycowsay")
+    assert pipx_metadata.main_package.package == "pycowsay"
+    assert pipx_metadata.main_package.package_or_url == "pycowsay"
+    assert pipx_metadata.main_package.pip_args == []
+    assert pipx_metadata.main_package.include_dependencies == False  # noqa: E712
+    assert pipx_metadata.main_package.include_apps == True  # noqa: E712
+    if pipx.constants.WINDOWS:
+        assert pipx_metadata.main_package.apps == ["pycowsay", "pycowsay.exe"]
+        assert pipx_metadata.main_package.app_paths == [
+            pipx_venvs_dir / "pycowsay" / "Scripts" / "pycowsay.exe"
+        ]
+    else:
+        assert pipx_metadata.main_package.apps == ["pycowsay"]
+        assert pipx_metadata.main_package.app_paths == [
+            pipx_venvs_dir / "pycowsay" / "bin" / "pycowsay"
+        ]
+    assert pipx_metadata.main_package.apps_of_dependencies == []
+    assert pipx_metadata.main_package.app_paths_of_dependencies == {}
+    assert pipx_metadata.main_package.package_version != ""
+
+    del pipx_metadata
+
+    # TODO 20191103: need simpler non-gcc-compiling package besides black!
+    run_pipx_cli(["inject", "pycowsay", "black"])
+
+    pipx_metadata = PipxMetadata(pipx_venvs_dir / "pycowsay")
+    assert pipx_metadata.injected_packages["black"].package == "black"
+    assert pipx_metadata.injected_packages["black"].package_or_url == "black"
+    assert pipx_metadata.injected_packages["black"].pip_args == []
+    assert (
+        pipx_metadata.injected_packages["black"].include_dependencies == False
+    )  # noqa: E712
+    assert pipx_metadata.injected_packages["black"].include_apps == False  # noqa: E712
+    if pipx.constants.WINDOWS:
+        # order is not important, so we compare sets
+        assert isinstance(pipx_metadata.injected_packages["black"].apps, list)
+        assert set(pipx_metadata.injected_packages["black"].apps) == {
+            "black",
+            "black.exe",
+            "blackd",
+            "blackd.exe",
```

Comment:
Ah this is actually a bug in pipx. The windows versions shouldn't have the non-exe versions. Suggest putting a TODO with a link to https://github.com/pipxproject/pipx/issues/217.

Reply:
Done.